rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -173,9 +174,27 @@ static bool encl_create(int dev_fd, unsigned long bin_size,
 	secs->attributes = SGX_ATTR_MODE64BIT;
 	if (enclave_debug)
 		secs->attributes |= SGX_ATTR_DEBUG;
+	if (meta_data->attributes)
+		secs->attributes = meta_data->attributes & secs->attributes;
 
 	get_sgx_xfrm_by_cpuid(&xfrm);
 	secs->xfrm = xfrm;
+	if (meta_data->xfrm)
+		secs->xfrm = meta_data->xfrm & secs->xfrm;
```

Comment:
The xfrm from metadata area indicates the subjective requirement to enforce. So if any bit in xfrm from metadata is not available, an error should be raised.

```C
if (meta_data->xfrm) {
    if (secs->xfrm & meta_data->xfrm != meta_data->xfrm)
        err_exit("...");
}
```